### PR TITLE
refactor: virtual block type and storage

### PR DESCRIPTION
**Pull request by radkomih**
This PR only contains refactoring of the type and storage ops. 
Handling the case of missing preconfirmation block will be in a separate PR.

### CODE REVIEW

File: eth/taiko_api_backend.go [hunk 1]
```diff
@@ -192,42 +193,29 @@ func compress(txListBytes []byte) ([]byte, error) {
...
-
-	if (hash == common.Hash{}) && number == nil {
-		return HashAndNumber{}
-	}
-
-	return HashAndNumber{hash, *number}
-}
-
-// GetPendingVirtualBlock ...
 func (s *TaikoAPIBackend) GetPendingVirtualBlock() HashAndNumber {
```
Comment (vladimir-trifonov) [inline eth/taiko_api_backend.go:202]: Should be:
```
func (s *TaikoAPIBackend) GetPendingPreconfBlock() HashAndNumber {
```
right?

File: eth/taiko_api_backend.go [hunk 2]
```diff
@@ -192,42 +193,29 @@ func compress(txListBytes []byte) ([]byte, error) {
...
 }
 
-// UpdatePreconfirmedVirtualBlock ...
-func (s *TaikoAPIBackend) UpdatePreconfirmedVirtualBlock(blockHash common.Hash, blockNumber *math.HexOrDecimal256) bool {
-	rawdb.WritePreconfirmedVirtualBlock(s.eth.ChainDb(), blockHash, (*big.Int)(blockNumber))
-	return true
-}
-
-// UpdatePendingVirtualBlock ...
 func (s *TaikoAPIBackend) UpdatePendingVirtualBlock(blockHash common.Hash, blockNumber *math.HexOrDecimal256) bool {
```
Comment (vladimir-trifonov) [inline eth/taiko_api_backend.go:212]: same here:
```
UpdatePendingPreconfBlock
```

**Review by vladimir-trifonov — APPROVED** [review]

File: core/types/preconf_block_cursor.go [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package types
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// Cursor that keeps track of the latest virtual block used for preconfirmed txs
```
Comment (failfmi) [inline core/types/preconf_block_cursor.go:7]: nit: `PreconfBlockCursor keeps track...`

**Review by failfmi — APPROVED** [review]

File: eth/taiko_api_backend.go (hunk 1 above)
Comment (radkomih) [reply eth/taiko_api_backend.go:202]: yes, but I will change the naming as part of the other PR (also in the taiko-client)

**Pull request merged**